plugins/scaffolder/src/components/MultistepJsonForm/MultistepJsonForm.tsx

```diff
@@ -116,13 +120,53 @@ export const MultistepJsonForm = ({
   const [activeStep, setActiveStep] = useState(0);
   const [disableButtons, setDisableButtons] = useState(false);
   const errorApi = useApi(errorApiRef);
+  const featureFlagApi = useApi(featureFlagsApiRef);
+  const featureFlagKey = 'backstage:featureFlag';
+  const filterOutProperties = (schema: JsonObject): JsonObject => {
+    const removedPropertyKeys: Array<string> = [];
+    if (schema.properties) {
+      schema.properties = Object.fromEntries(
+        Object.entries(schema.properties).filter(([key, value]) => {
+          if (value[featureFlagKey]) {
+            if (featureFlagApi.isActive(value[featureFlagKey])) {
+              return true;
+            }
+            removedPropertyKeys.push(key);
+            return false;
+          }
+          return true;
+        }),
+      );
+
+      // remove the feature flag property key from required if they are not active
+      if (Array.isArray(schema.required) && removedPropertyKeys.length > 0) {
+        for (const property of removedPropertyKeys) {
+          const index = schema.required.findIndex(r => r === property);
+          if (index > -1) {
+            schema.required.splice(index, 1);
+          }
+        }
+      }
+    }
+    return schema;
+  };
+
+  const updatedSteps = steps.filter(s => {
+    const featureFlag = s.schema[featureFlagKey] as string;
+    if (featureFlag && !featureFlagApi.isActive(featureFlag)) {
```

Comment:
safer to skip the `as string` cast above, and instead check `if (typeof featureFlag === 'string' && ...` here